### DISCUSSION TITLE
Any idea on how to have a NaiveTime input ?

### DISCUSSION THREAD

**Discussion by hugo-b-r** [emilk/egui#4379] [Q&A]
I am writing an interface that requires to take time as input but i'm already storing it in chrono's NaiveTime format. Do you have any idea on how to do that? I thought something like two textedit separated by a ':' but how then could I change the NaiveTime value when the user changes the text ? and what if he types some text instead of numbers ?
Thanks in advance for your help !

**Comment by thmxv** [comment]
I suppose you are only interested in the hours and minutes parts of the `NaiveTime`. Not the seconds and milli/nano seconds parts of it.

Anyway. I recommend using 2 more variables (named `hours` and `minutes` of type `u32`) and use two `DragValue` widgets (with limited range of 0 to 23 and 0 to 59 and separated by a ":" label) to allow editing those 2 variables. To construct a `NaiveTime` value when either one of those variable is changed, you can use the [`Response::changed()`](https://docs.rs/egui/latest/egui/struct.Response.html#method.changed) method just like the `Response::clicked()` method for buttons.

**Reply by hugo-b-r** [reply]
Thanks for your help. I don't have time to look at it as of now but it looks great. Thanks again.
